### DISCUSSION TITLE
change a user's email with a token.

### DISCUSSION THREAD

**Discussion by TimDR-1356** [supabase/supabase#19056] [Questions]
I want to change a logged in user's email with the tokens he receives on the new and old email. This is the code i have now i use react.

```
  const { data, error } = await supabase.auth.updateUser({
        email: email
  });
```
When i run the above code the user gets a email with a token on his old email and one on his new email. This is my email template:

```
<h2>Confirm Change of Email</h2>

<p>Follow this link to confirm the update of your email from {{ .Email }} to {{ .NewEmail }}:</p>
<p>{{ .Token }}</p>
```

Now I think it should go something like this:

```
const verifyNewEmail = async (email, token) => {
    const { data, error } = await supabase.auth.verifyOtp({
      email: email,
      token: token,
      type: 'email'
    });
    return { data, error };
}

const resetEmailVerifyEmails = async(oldEmailToken, newEmailToken) =>{
  const {data: oldEmailData, error: oldEmailError} = await verifyNewEmail(oldEmail, oldEmailToken);
  const {data: newEmailData, error: newEmailError} = await verifyNewEmail(newEmail, newEmailToken);

   if(oldEmailError){
      showPopUpMessage(oldEmailError.message, "Warning", 4000);
    }

    if(newEmailError){
      showPopUpMessage(newEmailError.message, "Warning", 4000);
    }
}
```

When I run the above code I get the message: "user does not exist". Which is normal because there is no user with the new email. Does anyone here know how to do this? I looked in the documents but I couldn't find an answer there.

**Comment by GaryAustin1** [comment]
https://github.com/supabase/gotrue-js/issues/798
Seems related with some suggestions... but also a support request at end.  
Seems there was a fix in April so you need > v2.60.5 gotrue on your instance.

**Comment by TimDR-1356** [comment]
Okay, I know what I did wrong i had to change type: "email"  to type: "email_change"
